src/components/ActivityDetail.tsx

```diff
@@ -0,0 +1,654 @@
+                    : i18n.t("activity.transaction_details.lightning_send")}
+                <img src={bolt} alt="lightning bolt" class="h-4 w-4" />
+            </div>
+            <ActivityAmount
+                size="2xl"
+                satsWeight="semibold"
+                fiatWeight="normal"
+                center
+                amount={props.info.amount_sats?.toString() ?? "0"}
+                price={state.price}
+                positive={props.info.inbound}
+            />
+        </div>
+    );
+}
+
+function OnchainHeader(props: { info: OnChainTx; kind?: HackActivityType }) {
+    const i18n = useI18n();
+    const [state, _actions] = useMegaStore();
+
+    const isSend = () => {
+        return props.info.sent > props.info.received;
+    };
+
+    const amount = () => {
+        if (isSend()) {
+            return (props.info.sent - props.info.received).toString();
+        } else {
+            return (props.info.received - props.info.sent).toString();
+        }
+    };
+
+    return (
+        <div class="flex flex-col items-center gap-4">
+            <div class="flex flex-row items-center justify-center gap-[4px] font-normal">
+                {props.kind === "ChannelOpen"
+                    ? i18n.t("activity.transaction_details.channel_open")
+                    : props.kind === "ChannelClose"
+                    ? i18n.t("activity.transaction_details.channel_close")
+                    : isSend()
+                    ? i18n.t("activity.transaction_details.onchain_send")
+                    : i18n.t("activity.transaction_details.onchain_receive")}
+                <Switch>
+                    <Match
+                        when={
+                            props.kind === "ChannelOpen" ||
+                            props.kind === "ChannelClose"
+                        }
+                    >
+                        <img src={shuffle} alt="swap" class="h-4 w-4" />
+                    </Match>
+                    <Match when={true}>
+                        <img src={chain} alt="blockchain" class="h-4 w-4" />
+                    </Match>
+                </Switch>
+            </div>
+            <Show when={props.kind !== "ChannelClose"}>
+                <ActivityAmount
+                    size="2xl"
+                    satsWeight="semibold"
+                    fiatWeight="normal"
+                    center
+                    amount={amount() ?? "0"}
+                    price={state.price}
+                    positive={!isSend()}
+                />
+            </Show>
+        </div>
+    );
+}
+
+export const KeyValue: ParentComponent<{ key: string; bigGap?: boolean }> = (
+    props
+) => {
+    return (
+        <li
+            class="flex items-start justify-between"
+            classList={{
+                "gap-6": !props.bigGap,
+                "gap-24": props.bigGap
+            }}
+        >
+            <span class="max-w-[75px] text-sm font-semibold uppercase text-gray-400">
```

Comment:
I know this is a nit but don't want to forget: you want `m-grey-400` here. `gray-400` is a tailwind color and has some blue in it. WAY different lol